### ISSUE TITLE
Assignment 4 Part I: Test Set Error is Lower than the Training Set Error

### ISSUE THREAD

**Issue by violetwhitney** [dmc-2016/dmc#79]
In Part 1 of the Assignment I adjusted the Hyper Parameters to simplify the model, in order to not overfit the model to the training set. 

_My Hyper Parameters_
![screen shot 2016-10-08 at 9 39 52 am](https://cloud.githubusercontent.com/assets/14256456/19239769/8e1053f8-8ed5-11e6-8401-73855036dc7f.png)

However, in my case the test set error is lower than the training set error, rather than the other way around. 

_My Error Rate_
![screen shot 2016-10-08 at 9 39 12 am](https://cloud.githubusercontent.com/assets/14256456/19239780/97f61132-8ed5-11e6-9c4f-44b7da1e9485.png)

In all other examples, the training set error is more accurate (lower) than the test set. Am I doing something wrong?  

_Other Example of Error Rate_
![screen shot 2016-10-08 at 9 51 01 am](https://cloud.githubusercontent.com/assets/14256456/19239785/9f1c938c-8ed5-11e6-98b4-159e8b3cf008.png)

**Comment by mengchantang** [comment]
I have the same problem on both parts of the assignment. This is quite counterintuitive. 
![image](https://cloud.githubusercontent.com/assets/19785356/19284242/16c8dd80-8fc3-11e6-8d26-cc9f172473c8.png)
![image](https://cloud.githubusercontent.com/assets/19785356/19284830/7b7b39c4-8fc5-11e6-9b2d-2e102446fec2.png)

**Comment by danilnagy** [comment]
Hey guys, these area great questions and valid concerns. The curves I drew in class represent the typical case, where performance over the test set will be somewhat worse than on the training set. This makes sense because, since the algorithm is learning from the training set it is always inherently biased to perform better on that set than on one it's never seen before.

However, this is definitely not a rule, and from time to time you do see cases where you actually get better performance on the test set. This usually doesn't mean anything specific about your model setup, and does not imply poor performance. However, it is more likely to happen with smaller data sets, such as the ones we are using for demonstrations in the class. This is because any given model is more likely to 'guess right' on a small set of samples even if the model is not that good.

In the real world you would not typically use such advanced algorithms for such small sets of data, and are thus less likely to see this issue in real world cases where you might be using several orders of magnitude more data.

Hope this helps!

**Comment by violetwhitney** [comment]
Yep! Thanks!
